### ISSUE TITLE
Access level when not logged-in

### ISSUE THREAD

**Issue by bramley** [phpList/phplist3#759]
Trying to install phplist with php 8 is showing a current problem.

php 8 compares a number and a string only when the string looks like a number see https://www.php.net/manual/en/language.operators.comparison.php whereas php 7 converts the string to a number before comparing.

The problem is the use of function accessLevel() which returns a string or 0 when an admin is not logged-in.
```
function accessLevel($page)
{
    global $tables, $access_levels;
    if (isSuperUser()) {
        return 'all';
    }
    if (!isset($_SESSION['adminloggedin'])) {
        return 0;
    }
    if (!is_array($_SESSION['logindetails'])) {
        return 0;
    }

    //# for non-supers we only allow owner views
    //# this is likely to need tweaking
    return 'owner';
}
```
That return value is usually used to compare with specific privilege values such as

    if ($access == 'owner' || $access == 'all' || $access == 'view') {
With php 7 those comparisons are true when accessLevel() returned 0.

Function PageLink2() and related functions use accessLevel() and return a page link even when an admin is not logged-in because of the "loose" comparison. I guess that is not really intended and should be changed.

With php 8 strict comparison PageLink2() will always return an empty string when an admin is not logged-in. That breaks some links or buttons in the initialisation code for a new installation, see https://discuss.phplist.org/t/web-installation-error-initialize-database-missing/7326 where the "continue" link is now omitted.

**Comment by michield** [comment]
Ah, no, basically it worked because of the loose comparison in PHP7, but changing that will break it in both. I only now read it more thoroughly. I guess we need to rewrite that continue link to not use PageLink2. It's quite an edge case.

**Comment by bramley** [comment]
Yes, I think having accesslevel() return an empty string or even "none" to be clearer, and changing all non-logged in code to not use PageLink2 and its equivalents. There is at least one link and one button in the phplist installation processing, maybe others elsewhere.

**Comment by michield** [comment]
Interesting. This is probably also the cause for the PHP8 Behat tests failing

**Comment by michield** [comment]
Resolved with #777
